app/queries/reasons_for_rejection_count_query.rb

```diff
@@ -0,0 +1,99 @@
+class ReasonsForRejectionCountQuery
+  THIS_MONTH = 'this_month'.freeze
+  BEFORE_THIS_MONTH = 'before_this_month'.freeze
+
+  Result = Struct.new(:all_time, :this_month, :sub_reasons)
+
+  def reason_counts
+    rows = ActiveRecord::Base.connection.exec_query(
+      reason_counts_sql,
+      'SQL',
+      [[nil, Time.zone.now.beginning_of_month]],
+    ).to_a
+
+    to_results(rows)
+  end
+
+  def sub_reason_counts
+    results = reason_counts
+
+    rows = ActiveRecord::Base.connection.exec_query(
+      sub_reason_counts_sql,
+      'SQL',
+      [[nil, Time.zone.now.beginning_of_month]],
+    ).to_a
+
+    add_sub_results(results, rows)
+  end
+
+private
+
+  MAPPING = {
```

Comment:
Is this subreasons mapped to top-level reasons? Are some missing, eg - `candidate_behaviour_what_to_improve`?

Reply:
I don't think we need to include `candidate_behaviour_what_to_improve` here because it's the free text question that used to candidates to elaborate if they select _other_ for `candidate_behaviour_what_did_the_candidate_do`. This list is limited to the questions that have sub-reasons as answers.

Reply:
Would something like `SUBREASONS_TO_TOP_LEVEL_REASONS` be a little clearer?